File: alogrithm/sort/shellsort/shsort.py
def shsort(array):
    interval = len(array)/2
    while interval >= 1:
        i = interval
        while i < len(array):
            if array[i] < array[i-interval]:
                j=i-interval
                temp=array[i]
                while j >=0 and array[j] >temp:
                    array[j+interval]=array[j]
                    j-=interval
                array[j+interval]=temp
            i+=1
        interval/=2

    return array


if __name__=="__main__":

    array=[3,6,9,5,7,4,8,2,1]
    print shsort(array)


         

File: alogrithm/sort/shellsort/shsort.c
#include<stdio.h>

void shsort(int *array,int n)
{
	int interval = n/2;
	int i,j;
	int temp;
	while(interval>=1)
	{
		i = interval;
		while(i<n)
		{
			if(array[i-interval]>array[i])
			{
				j = i-interval;
				temp = array[i];
				while(j>=0&&array[j]>temp)
				{
					array[j+interval] = array[j];
					j -= interval;
				}
				array[j+interval] = temp;
			}
		i++;
		}
		interval /=2;
	}
}


int main(void)
{
	int array[] = {3,6,2,8,4,7,5,9};
	shsort(array,8);
	int i;
	for(i=0;i<8;i++)
		printf("%d ",array[i]);
	return 0;
}





